### ISSUE TITLE
Remove all items within a range

### ISSUE THREAD

**Issue by dipu-bd** [alexbol99/flatten-interval-tree#38]
There is a method to remove a single item from the tree. 

```
tree.remove(key, value)
```

I was wondering is there any method to remove all items within a range?

**Comment by aphillipo** [comment]
Would you want overlapping or fully within the range? You can probably do it with tree.map() which returns a new tree. It's definitely possible but I guess that would be quite inefficient as you've have to traverse the whole tree making the operation O(n).

**Comment by dipu-bd** [comment]
Yes, It would not be efficient. Deleting a range can be done in O(log n)
